src/test/java/com/flightstats/hub/model/ChannelConfigTest.java

```diff
@@ -169,14 +171,47 @@ void testEnforceChannelRetention() {
     }
 
     @Test
-    void testMutableTime() {
-        ChannelConfig defaults = ChannelConfig.builder().name("defaults").build();
+    void testWillDefaultMutableChannelToSingle() {
         DateTime mutableTime = TimeUtil.now();
-        ChannelConfig channelConfig = defaults.toBuilder().mutableTime(mutableTime).build();
+        ChannelConfig historical = ChannelConfig.builder()
+                .mutableTime(mutableTime)
+                .name("historical").build();
+        assertEquals(historical.getMutableTime(), mutableTime);
+        assertEquals(historical.getStorage(), ChannelType.SINGLE.name());
+    }
+
+    @Test
+    void testCanUpdateToMutableTimeIfUpdateStorage() {
+        ChannelConfig defaults = ChannelConfig.builder()
+                .storage(ChannelType.SINGLE.name())
+                .name("defaults").build();
+
+        DateTime mutableTime = TimeUtil.now();
+        ChannelConfig channelConfig = defaults.toBuilder().mutableTime(mutableTime).storage(ChannelType.SINGLE.name()).build();
+        assertEquals(mutableTime, channelConfig.getMutableTime());
+    }
+
+    @Test
+    void testMutableTimeFailOnUpdateWithoutUpdateStorage() {
```

Comment:
While this is a reasonable test, I think it's more likely that we'd start with a SINGLE mutable time channel and then try to change storage type to BATCH. I believe we already prevent changing between a historical channel and something with maxItems or a TTL... Maybe add a test for the first case as well?

Reply:
this and the test case you suggested have moved to the channel validator test where the error now originates